src/protocols/shell.rs

```diff
@@ -0,0 +1,292 @@
+//! Shell Protocol
+//!
+//! Provides shell services to UEFI applications.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x6302d008,
+    0x7f9b,
+    0x4f30,
+    0x87,
+    0xac,
+    &[0x60, 0xc9, 0xfe, 0xf5, 0xda, 0x4e],
+);
+
+pub type ShellFileHandle = *mut core::ffi::c_void;
```

Comment:
This is actually `SHELL_FILE_HANDLE` in the spec? Why does this not have the `EFI_*` prefix? *sigh*

If it was `EFI_SHELL_FILE_HANDLE`, we would call it `FileHandle` (drop the `shell` prefix). What do you think, should we rename to `FileHandle` to be more consistent?

Reply:
I think it would be fine to rename it to `FileHandle` since it is not used in any other Shell Protocols.